llm_studio/src/utils/modeling_utils.py

```diff
@@ -75,15 +77,27 @@ def save_checkpoint(model: torch.nn.Module, path: str, cfg: Any):
         Dictionary with all the keys to save
     """
 
-    model = unwrap_model(model)
-
-    if hasattr(cfg.training, "lora") and cfg.training.lora:
-        model.backbone.save_pretrained(path)
-
-    checkpoint = {"model": model.state_dict()}
-
-    if path is not None:
-        torch.save(checkpoint, os.path.join(path, "checkpoint.pth"))
+    if cfg.environment.use_deepspeed:
+        if path is not None:
+            # gather model params from all ranks
+            model.save_checkpoint(os.path.join(path, "ds_checkpoint"))
+            if cfg.environment._local_rank == 0:
+                # load to cpu
+                state_dict = get_fp32_state_dict_from_zero_checkpoint(
+                    os.path.join(path, "ds_checkpoint")
+                )
+                # save as normal checkpoint that can be loaded by `load_state_dict`
+                checkpoint = {"model": state_dict}
+                torch.save(checkpoint, os.path.join(path, "checkpoint.pth"))
+                shutil.rmtree(os.path.join(path, "ds_checkpoint"))
```

Comment:
where are we saving the LoRA adapter?

Reply:
I dont' have good idea to save it as lora.
I tried to do the following but get stuck when creating model_for_save.
Do you have any idea on it? @pascal-pfeiffer 

<img width="596" alt="image" src="https://github.com/h2oai/h2o-llmstudio/assets/10557602/af7e84ff-b220-4ea5-b9b3-a9e39e57ecf7">

Reply:
fix in [7b78538](https://github.com/h2oai/h2o-llmstudio/pull/288/commits/7b7853868f19d6e483462e20b5c14f0767535415)